core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -32,6 +50,7 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
 
     sealed class ResolveResult {
```

Comment:
> That somewhat maps to current API in a following way

This way is fine too but in (the most useful) case of success, you have to use `dependencies!!`, which is unfortunate.

> Not sure why we should avoid hierarchies. Can you explain your reasoning a bit?

Because the two subclasses here are not very different, and IMHO they add more complexity than value. A data class is easier to read and use than a sealed hierarchy of two nested sealed classes. Compare
```
val result = resolver.resolve(...)
if (result is DependenciesResolver.ResolutionResult.Success) {
    ...
}
```

```
val result = resolver.resolve(...)
if (result.isSuccess) {
    ...
}
```